test/slp_server_test.rb

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+
+Yast.import "SlpServer"
+
+RSpec.describe "Yast::SlpServer" do
+  subject(:slp_server) { Yast::SlpServer }
+
+  let(:fixtures) { File.join(FIXTURES_PATH, "valid") }
+
+  describe "#Read" do
+    let(:as_root) { true }
+    let(:installed_packages) { true }
+
+    before do
+      allow(Yast::Builtins).to receive(:sleep)
+      allow(Yast::Confirm).to receive(:MustBeRoot).and_return(as_root)
+      allow(Yast::Progress).to receive(:NextStage)
+      allow(Yast::NetworkService).to receive(:RunningNetworkPopup).and_return(true)
+      allow(Yast::SuSEFirewall).to receive(:Read)
+      allow(slp_server).to receive(:installed_packages).and_return(installed_packages)
+    end
+
+    around do |example|
+      change_scr_root(fixtures) do
+        example.run
+      end
+    end
+
+    context "given a valid configuration" do
+      it "updates SLP config and returns true" do
+        old_settings = slp_server.slp_config.clone
+        expect(slp_server.Read).to eq(true)
+        expect(slp_server.slp_config)
+          .to eq(old_settings.merge("net.slp.useIPv4" => "false"))
+      end
+    end
+
+    context "given a broken configuration (bsc#878892)" do
+      let(:fixtures) { File.join(FIXTURES_PATH, "broken") }
+
+      it "updates SLP config and returns true" do
+        old_settings = slp_server.slp_config.clone
+        expect(slp_server.Read).to eq(true)
+        expect(slp_server.slp_config)
+          .to eq(old_settings.merge(
+            "net.slp.isBroadcastOnly" => "true",
+            "net.slp.DAAddresses" => " ",
+            "net.slp.isDA" => "false"
+          ))
+      end
+
+      # '; net.slp.DAHeartBeat' was intepreted as
+      # a param named ';' with value 'net.slp.DAHeartBeat'
```

Comment:
please add reference to bug here, so you can check more details and reasons in fture